src/raw.rs

```diff
@@ -11,10 +11,294 @@
+        Scan::new(
+            self.rpc(),
+            ScanInner::new(range.into_bounds(), limit.into().unwrap_or(u32::MAX)),
+        )
+    }
+
+    /// Create a new [`BatchScan`](struct.BatchScan.html) request.
+    ///
+    /// Once resolved this request will result in a set of scanners over the given keys.
+    ///
+    /// If not passed a `limit` parameter, it will default to `u32::MAX`.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Key, Config, raw::Client};
+    /// # use futures::Future;
+    /// # let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let inclusive_range1 = "TiDB"..="TiKV";
+    /// let inclusive_range2 = "TiKV"..="TiSpark";
+    /// let iterable = vec![inclusive_range1, inclusive_range2];
+    /// let req = connected_client.batch_scan(iterable, 2);
+    /// let result = req.wait();
+    /// ```
+    pub fn batch_scan(
+        &self,
+        ranges: impl IntoIterator<Item = impl KeyRange>,
+        each_limit: impl Into<Option<u32>>,
+    ) -> BatchScan {
+        BatchScan::new(
+            self.rpc(),
+            BatchScanInner::new(
+                ranges.into_iter().map(KeyRange::into_keys).collect(),
+                each_limit.into().unwrap_or(u32::MAX),
+            ),
+        )
+    }
+
+    /// Create a new [`DeleteRange`](struct.DeleteRange.html) request.
+    ///
+    /// Once resolved this request will result in the deletion of all keys over the given range.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Key, Config, raw::Client};
+    /// # use futures::Future;
+    /// # let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let inclusive_range = "TiKV"..="TiDB";
+    /// let req = connected_client.delete_range(inclusive_range);
+    /// let result: () = req.wait().unwrap();
+    /// ```
+    pub fn delete_range(&self, range: impl KeyRange) -> DeleteRange {
+        DeleteRange::new(self.rpc(), DeleteRangeInner::new(range.into_keys()))
+    }
+}
+
+/// An unresolved [`Client`](struct.Client.html) connection to a TiKV cluster.
+///
+/// Once resolved it will result in a connected [`Client`](struct.Client.html).
+///
+/// ```rust,no_run
+/// use tikv_client::{Config, raw::{Client, Connect}};
+/// use futures::Future;
+///
+/// let connect: Connect = Client::new(&Config::default());
+/// let client: Client = connect.wait().unwrap();
+/// ```
+pub struct Connect {
+    config: Config,
+}
+
+impl Connect {
+    fn new(config: Config) -> Self {
+        Connect { config }
+    }
+}
+
+impl Future for Connect {
+    type Item = Client;
+    type Error = Error;
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let config = &self.config;
+        let rpc = Arc::new(RpcClient::connect(config)?);
```

Comment:
Is `RpcClient::connect` synchronous? From reading `rpc::client` it's not obviously asynchronous.

Reply:
It is synchronous at this time. I thought it might be good to leave to possibility to support asynchronous connect later on. That way people can implement proxy like applications that support connect multiple TiKV cluster on the fly without blocking the core. Since it is probably something that's unlikely to happen in real life, if changing connect to blocking style is easier to understand and use, let's change it.

BTW: Thank you so much for all the suggestions, let me take a look and fix them.

Reply:
It's fine for now, but istm that ideally everything running on an async event loop should be non-blocking. It's probably worth noting in the doc comments that this future is not asynchronous.